common/functions/src/scalars/strings/locate.rs

```diff
@@ -0,0 +1,248 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt;
+
+use common_datavalues::prelude::*;
+use common_exception::Result;
+use itertools::izip;
+
+use crate::scalars::function_factory::FunctionDescription;
+use crate::scalars::function_factory::FunctionFeatures;
+use crate::scalars::Function;
+
+#[derive(Clone)]
+pub struct LocateFunction {
+    display_name: String,
+}
+
+impl LocateFunction {
+    pub fn try_create(display_name: &str) -> Result<Box<dyn Function>> {
+        Ok(Box::new(LocateFunction {
+            display_name: display_name.to_string(),
+        }))
+    }
+
+    pub fn desc() -> FunctionDescription {
+        FunctionDescription::creator(Box::new(Self::try_create))
+            .features(FunctionFeatures::default().deterministic())
+    }
+}
+
+impl Function for LocateFunction {
+    fn name(&self) -> &str {
+        &*self.display_name
+    }
+
+    fn num_arguments(&self) -> usize {
+        0
+    }
+
+    fn variadic_arguments(&self) -> Option<(usize, usize)> {
+        Some((2, 3))
+    }
+
+    fn nullable(&self, _input_schema: &DataSchema) -> Result<bool> {
+        Ok(true)
+    }
+
+    fn return_type(&self, _args: &[DataType]) -> Result<DataType> {
+        Ok(DataType::UInt64)
+    }
+
+    fn eval(&self, columns: &DataColumnsWithField, input_rows: usize) -> Result<DataColumn> {
+        let ss_column = columns[0].column().cast_with_type(&DataType::String)?;
+        let s_column = columns[1].column().cast_with_type(&DataType::String)?;
+        let p_column = if columns.len() == 3 {
+            columns[2].column().cast_with_type(&DataType::UInt64)?
+        } else {
+            DataColumn::Constant(DataValue::UInt64(Some(1)), input_rows)
+        };
+
+        let r_column: DataColumn = match (ss_column, s_column, p_column) {
+            (
+                DataColumn::Constant(DataValue::String(ss), _),
+                DataColumn::Constant(DataValue::String(s), _),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(ss), Some(s), Some(p)) = (ss, s, p) {
+                    DataColumn::Constant(DataValue::UInt64(Some(find_at(&s, &ss, &p))), input_rows)
+                } else {
+                    DataColumn::Constant(DataValue::Null, input_rows)
+                }
+            }
+            (
+                DataColumn::Constant(DataValue::String(ss), _),
+                DataColumn::Array(s_series),
+                DataColumn::Constant(DataValue::UInt64(p), _),
+            ) => {
+                if let (Some(ss), Some(p)) = (ss, p) {
+                    let s_array = s_series.string()?;
+                    DFUInt64Array::new_from_iter_validity(
+                        s_array.into_no_null_iter().map(|s| find_at(s, &ss, &p)),
+                        combine_validities(s_array.inner().validity(), None),
```

Comment:
Why not just use `s_array.inner().validity().clone()` ?

Reply:
`combine_validities` do it internally, anyway changed to `inner().validity().cloned()`...